### PR TITLE
Refs #28428 -- Made FileField.upload_to support pathlib.Path.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: tests/file_storage/models.py [hunk 1]
```diff
@@ -31,8 +32,13 @@ def random_upload_to(self, filename):
         # to make sure it only gets called once.
         return '%s/%s' % (random.randint(100, 999), filename)
 
+    def pathlib_upload_to(self, filename):
+        return Path('bar') / filename
```
Comment (felixxm) [inline tests/file_storage/models.py]: Can we move tests for callable that returns `pathlib.Path` to a separate commit? It works before this change.

**Comment by felixxm** [comment]
@claudep Thanks for this patch :+1: I split commits, rebased from master, added release notes and pushed minor edits.

I wonder if this patch will completely fix ticket 28428, because it's about "adding support for Pathlib objects in django.core.storage". Maybe we should change "Fixed" to "Refs" or change a ticket description :thinking: What do you think?

**Comment by felixxm** [comment]
buildbot

**Comment by claudep** [comment]
Thanks for excellent polishing.
You are right in that this patch doesn't guarantee pathlib support in all storage.py. Probably the next step would be to support a `Path` in `MEDIA_ROOT`. We can keep the ticket open but it might not be easy to determine when it will really be fixed.

**Comment by felixxm** [comment]
@claudep Thanks :+1: Let's keep the ticket open.

**Pull request merged**